### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by chilampoon**
- Rewrite Roxygen docs
- Update vignette
- Add a `multiObjects()` function -- https://github.com/reactome/ReactomeGraph4R/issues/19#issue-723407109
- Shorten lines...
- XQuartz, package installation issues fixed

### CODE REVIEW

**Comment by chilampoon** [comment]
BiocCheck (Bioconductor check) passed:
``` r
BiocCheck::BiocCheck("~/pkg/ReactomeGraph4R")
#> This is BiocCheck version 1.24.0. BiocCheck is a work in progress.
#> Output and severity of issues may change. Installing package...
#> Is the url 'http://localhost:7474'? (Yes/no/cancel) 
#> Does Neo4J require authentication? (yes/No/cancel)
#> * Checking Package Dependencies...
#> * Checking if other packages can import this one...
#> * Checking to see if we understand object initialization...
#> * Checking for deprecated package usage...
#> * Checking for remote package usage...
#> * Checking version number...
#> * Checking version number validity...
#>     Package version 0.99.0; pre-release
#> * Checking R Version dependency...
#> * Checking package size...
#>         Skipped... only checked on source tarball
#> * Checking individual file sizes...
#> * Checking biocViews...
#> * Checking that biocViews are present...
#> * Checking package type based on biocViews...
#>     Software
#> * Checking for non-trivial biocViews...
#> * Checking that biocViews come from the same category...
#> * Checking biocViews validity...
#> * Checking for recommended biocViews...
#> * Checking build system compatibility...
#> * Checking for blank lines in DESCRIPTION...
#> * Checking if DESCRIPTION is well formatted...
#> * Checking for whitespace in DESCRIPTION field names...
#> * Checking that Package field matches directory/tarball name...
#> * Checking for Version field...
#> * Checking for valid maintainer...
#> * Checking DESCRIPTION/NAMESPACE consistency...
#> * Checking vignette directory...
#>     This is a software package
#> * Checking library calls...
#> * Checking for library/require of ReactomeGraph4R...
#> * Checking coding practice...
#> * Checking parsed R code in R directory, examples, vignettes...
#> * Checking function lengths...................
#>     * NOTE: Recommended function length <= 50 lines.
#>       There are 4 functions > 50 lines.
#>       The longest 5 functions are:
#>         matchObject() (R/queries.R, line 62): 79 lines
#>         .basicErrMsgs() (R/errors.R, line 2): 61 lines
#>         matchReferrals() (R/queries.R, line 445): 54 lines
#>         .parse_row() (R/data.parsing.R, line 53): 52 lines
#>         .WHERE() (R/cyphers.R, line 19): 46 lines
#> * Checking man page documentation...
#>     * NOTE: Consider adding runnable examples to the following man
#>       pages which document exported objects:
#>       unnestListCol.Rd
#>     * NOTE: Usage of dontrun{} / donttest{} found in man page examples.
#>       17% of man pages use one of these cases.
#>       Found in the following files:
#>         matchReferrals.Rd
#>         unnestListCol.Rd
#> * Checking package NEWS...
#> * Checking unit tests...
#> * Checking skip_on_bioc() in tests...
#> * Checking formatting of DESCRIPTION, NAMESPACE, man pages, R source,
#>   and vignette source...
#>     * NOTE: Consider shorter lines; 103 lines (4%) are > 80 characters
#>       long.
#>     First 6 lines:
#>       R/cyphers.R:11     clause <- paste0("MATCH p", list.idx, " = ", clause)...
#>       R/cyphers.R:31     # can't fetch data if adding Reactome/PubMed as data...
#>       R/cyphers.R:40   # complete WHERE clause by adding filter arguments (eg...
#>       R/cyphers.R:50       add <- paste0(node, '.speciesName = "', .matchSpec...
#>       R/cyphers.R:56       tmp <- ifelse(grepl("^[0-9]+$", filters[[filter]])...
#>       R/cyphers.R:61     clause <- ifelse(!grepl(node, clause), paste0(clause...
#>     * NOTE: Consider multiples of 4 spaces for line indents, 655
#>       lines(28%) are not.
#>     First 6 lines:
#>       R/cyphers.R:6   final.clause <- ""
#>       R/cyphers.R:7   
#>       R/cyphers.R:8   # glue clauses in the list
#>       R/cyphers.R:9   for (list.idx in seq(1, length(clause.list))) {
#>       R/cyphers.R:13   }
#>       R/cyphers.R:14   final.clause
#>     See http://bioconductor.org/developers/how-to/coding-style/
#>     See FormatR package:
#>       https://cran.r-project.org/web/packages/formatR/index.html
#> * Checking if package already exists in CRAN...
#> * Checking for bioc-devel mailing list subscription...
#>     * NOTE: Cannot determine whether maintainer is subscribed to the
#>       bioc-devel mailing list (requires admin credentials).  Subscribe
#>       here: https://stat.ethz.ch/mailman/listinfo/bioc-devel
#> * Checking for support site registration...
#>     Maintainer is registered at support site.
#> 
#> 
#> Summary:
#> ERROR count: 0
#> WARNING count: 0
#> NOTE count: 6
#> For detailed information about these checks, see the BiocCheck
#> vignette, available at
#> https://bioconductor.org/packages/3.11/bioc/vignettes/BiocCheck/inst/doc/BiocCheck.html#interpreting-bioccheck-output
#> $error
#> character(0)
#> 
#> $warning
#> character(0)
#> 
#> $note
#> [1] "Recommended function length <= 50 lines."                                                                                                                                          
#> [2] "Consider adding runnable examples to the following man pages which document exported objects:"                                                                                     
#> [3] "Usage of dontrun{} / donttest{} found in man page examples."                                                                                                                       
#> [4] "Consider shorter lines; 103 lines (4%) are > 80 characters long."                                                                                                                  
#> [5] "Consider multiples of 4 spaces for line indents, 655 lines(28%) are not."                                                                                                          
#> [6] "Cannot determine whether maintainer is subscribed to the bioc-devel\nmailing list (requires admin credentials).  Subscribe here:\nhttps://stat.ethz.ch/mailman/listinfo/bioc-devel"
```

<sup>Created on 2020-10-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Review by SolomonShorser-OICR — APPROVED** [review]

File: R/cyphers.R [hunk 1]
```diff
@@ -6,7 +6,7 @@
   final.clause <- ""
   
   # glue clauses in the list
-  for (list.idx in 1:length(clause.list)) {
+  for (list.idx in seq(1, length(clause.list))) {
```
Comment (cookersjs) [inline R/cyphers.R:9]: What was the reason for changing to `seq`?

File: R/queries.R [hunk 2]
```diff
@@ -106,33 +132,101 @@ matchObject <- function(id=NULL, displayName=NULL, schemaClass=NULL, species=NUL
...
+    # use doParallel
+    cl <- makeCluster(cluster)
+    registerDoParallel(cl)
+    
+    # combine dataframes
+    dfcomb <- function(...) {
+      rbindlist(list(...), fill=TRUE)
+    }
+    
+    id <- NULL # prevent note in R CMD Check
```
Comment (cookersjs) [inline R/queries.R:182]: Could you explain this comment?

File: R/queries.R [hunk 3]
```diff
@@ -106,33 +132,101 @@ matchObject <- function(id=NULL, displayName=NULL, schemaClass=NULL, species=NUL
...
+#' The argument "depth" is used to describe the "variable length relationships" 
+#' in Neo4j, default is 1 (i.e. immediately connected); or you can set 
+#' `all.depth = TRUE` to retrieve the whole context.
 #' 
-#' @param event.id a stable or db id of an Event
+#' @param event.id stId/dbId of an Event
 #' @param event.displayName displayName of an Event
 #' @param species name or taxon id or dbId or abbreviation of specified species
-#' @param all.depth if set to `TRUE`, all RLEs connected to the given Event in all depths returned
+#' @param all.depth if set to `TRUE`, all RLE(s) connected to the given Event
```
Comment (cookersjs) [inline R/queries.R:211]: I think this has been changed to just 'depth'

File: R/queries.R [hunk 4]
```diff
@@ -299,30 +416,34 @@ matchReactionsInPathway <- function(event.id=NULL, event.displayName=NULL, speci
 
 #' MATCH biological referrals
 #' 
-#' This method retrieves Reactome objects that are connected with the given object 
-#' in a reverse relationship. For example, to find Pathways contained the given Reaction.
+#' This method retrieves Reactome objects that are connected with the given 
+#' object in a _reverse_ relationship. For example, to find Pathways contained
```
Comment (cookersjs) [inline R/queries.R:420]: Pathways 'containing' the given Reaction

File: vignettes/Introduction.Rmd [hunk 5]
```diff
@@ -28,11 +28,11 @@ With `ReactomeGraph4R`, you are able to fetch the interconnected data easily usi
...
 ## Setups
 
-Follow this [instruction](https://github.com/reactome/ReactomeGraph4R/blob/develop/README.md) to download and setup Reactome Graph Database.
+Follow this [instruction](https://github.com/reactome/ReactomeGraph4R/blob/develop/README.md) to download and setup the Reactome Graph Database.
 
-Then install `ReactomeGraph4R` package via CRAN:
+Then install `ReactomeGraph4R` package:
 ```{r install, eval=FALSE}
-install.packages("ReactomeGraph4R")
+devtools::install_github("reactome/ReactomeGraph4R")
```
Comment (cookersjs) [inline vignettes/Introduction.Rmd:35]: Is `remotes::install_github("reactome/ReactomeGraph4R")` a valid way to install this package, like in the ContentService package?

File: R/cyphers.R (hunk 1 above)
Comment (chilampoon) [reply R/cyphers.R:9]: `1:...` will trigger the warning/note in biocCheck

File: R/queries.R (hunk 3 above)
Comment (chilampoon) [reply R/queries.R:211]: There is another parameter 'depth' for the number of depths

File: vignettes/Introduction.Rmd (hunk 5 above)
Comment (chilampoon) [reply vignettes/Introduction.Rmd:35]: Yes

File: R/queries.R (hunk 2 above)
Comment (chilampoon) [reply R/queries.R:182]: There would be a note without this line
```
no visible binding for global variable ‘id’ 
Undefined global functions or variables
```

Comment (cookersjs) [reply R/queries.R:182]: Ah got it, it wants it to be populated instead of just initialized

File: vignettes/Introduction.Rmd (hunk 5 above)
Comment (cookersjs) [reply vignettes/Introduction.Rmd:35]: I was just asking because I think for the CS package that the docs showed the two ways

**Review by cookersjs — APPROVED** [review]

**Comment by chilampoon** [comment]
Can I merge this branch? @jweiser

**Pull request merged**